sled-agent/src/services.rs

```diff
@@ -1015,57 +1092,146 @@ impl ServiceManager {
             services,
         };
 
-        self.ensure_switch_zone(Some(request)).await
+        self.ensure_zone(ZoneType::Switch, Some(request)).await
     }
 
     /// Ensures that no switch zone is active.
     pub async fn deactivate_switch(&self) -> Result<(), Error> {
-        self.ensure_switch_zone(None).await
+        self.ensure_zone(ZoneType::Switch, None).await
     }
 
-    // Forcefully initialize a switch zone.
+    /// Ensures that an NTP zone exists with the provided configuration.
+    pub async fn activate_ntp(
+        &self,
+        ntp_zone_ip: Option<Ipv6Addr>,
```

Comment:
`ntp_zone_ip` is provided as optional, but it always exists, right? *Should* it be optional?